src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsworkload/listener/ExtractPlacedEventListener.kt

```diff
@@ -2,15 +2,29 @@ package uk.gov.justice.digital.hmpps.hmppsworkload.listener
 
 import org.springframework.jms.annotation.JmsListener
 import org.springframework.stereotype.Component
+import uk.gov.justice.digital.hmpps.hmppsworkload.domain.OutOfDateReductions
+import uk.gov.justice.digital.hmpps.hmppsworkload.jpa.entity.WMTWorkloadOwnerEntity
 import uk.gov.justice.digital.hmpps.hmppsworkload.service.reduction.UpdateReductionService
+import uk.gov.justice.digital.hmpps.hmppsworkload.service.staff.RequestStaffCalculationService
 
 @Component
 class ExtractPlacedEventListener(
-  private val updateReductionService: UpdateReductionService
+  private val updateReductionService: UpdateReductionService,
+  private val requestStaffCalculationService: RequestStaffCalculationService
 ) {
 
   @JmsListener(destination = "hmppsextractplacedqueue", containerFactory = "hmppsQueueContainerFactoryProxy")
   fun processMessage(rawMessage: String) {
-    updateReductionService.updateOutOfDateReductionStatus()
+    val outOfDateReductions = updateReductionService.updateOutOfDateReductionStatus()
+
+    val staffChanged = getDistinctStaffChanged(outOfDateReductions)
+
+    staffChanged.forEach { workloadOwner ->
+      requestStaffCalculationService.requestStaffCalculation(workloadOwner.offenderManager.code, workloadOwner.team.code)
```

Comment:
should this go within the update reduction transaction? If we are unable to retrieve the grade of a staff member should that rollback the transaction?

Reply:
going to get rid of requesting grade as we already know the staff member's weekly hours

Reply:
Hmm.. really this whole method should be in a transaction. But if it fails then it will get resubmitted via the DLQ - will that not mean it will sort itself out anyway?